osu.Game.Rulesets.Osu/UI/ReplayAnalysisOverlay.cs

```diff
@@ -90,14 +111,23 @@ private void initialise()
                 FrameMarkers = new FrameMarkerContainer(),
             };
 
+            LoadComponentsAsync(newDrawables, drawables =>
+            {
+                InternalChildrenEnumerable = drawables;
+                loadState.Value = true;
+            }, generationCancellationSource.Token);
```

Comment:
I don't get any of this still. Why is this async-loading the content now? What is that needed for?

Reply:
The drawables were always async-loaded ([master](https://github.com/ppy/osu/blob/master/osu.Game.Rulesets.Osu/UI/ReplayAnalysisOverlay.cs#L125)). Peppy did that part, but I _assume_ it's because the drawables are reloaded every time the display length changes, so multiple changes in a short time-span would cause lag. 
I'm not familiar enough with the core logic behind lifetimes, but I am wondering, could `LifetimeEnd` in `AnalysisFrameEntry` be changed during replay (probably by a bindable), instead of reloading the entire overlay each time? I think that way the async-loading wouldn't be necessary.